src/main/java/org/apache/sysml/api/mlcontext/Script.java

```diff
@@ -325,6 +330,23 @@ public Script input(String name, Object value) {
 	 * @return {@code this} Script object to allow chaining of methods
 	 */
 	public Script in(String name, Object value, MatrixMetadata matrixMetadata) {
+		return in(name, value, matrixMetadata, false);
+	}	
+	/**
+	 * Register an input (parameter ($) or variable) with optional matrix
+	 * metadata.
+	 *
+	 * @param name
+	 *            name of the input
+	 * @param value
+	 *            value of the input
+	 * @param matrixMetadata
+	 *            optional matrix metadata
+	 * @param bFrame
+	 *            if input is of type frame
+	 * @return {@code this} Script object to allow chaining of methods
+	 */
+	public Script in(String name, Object value, MatrixMetadata matrixMetadata, boolean bFrame) {
```

Comment:
So one of the cool new features in the new `MLContext` is that the API seamlessly allows various input types and output types to flow through without the use of any extra boolean flags.  This makes the overall API really clean and great to use.  Could you explore removing the boolean flags and keeping the same new API style?  It's really nice from a user's perspective.

cc @deroneriksson Do you have any clever thoughts on this?  I know a lot of work went into this design, and it turned out great.